### DISCUSSION TITLE
Modeling a yoke

### DISCUSSION THREAD

**Discussion by fdinhSG** [magpylib/magpylib#674] [Q&A] [answered]
Hi! I want to model a simple yoke to direct the magnetic field. Something like this:

![image](https://github.com/magpylib/magpylib/assets/128691970/216bb209-8d1e-4f1c-89e2-bc09784d79a4)

I have no problem modeling the coil, but I'm stumped with the yoke. Do I simply model it as three `magpy.magnet.Cuboid` with (0,0,0) magnetization? Or is there a better way to represent it? How do I then define its magnetic reluctance?

Thanks!

**Comment by OrtnerMichael** [comment, marked as answer]
hi @fdinhSG

Magpylib only models permanent magnets (= magnetized hard magnetic materials) and coils. The solutions are based on no material response. This means that Magpylib is really good when you have magnets and coils around, but cannot be used to model a soft-magnetic body like your yoke.

The sub-package "magpylib-material-response" can do what you ask for, but its handling is more difficult, it requires meshing of the yoke so that the computation will be much slower.

The package is here

https://github.com/magpylib/magpylib-material-response

If you are interested but need help with this let me know !

**Reply by fdinhSG** [reply]
Thank you! I'll try experimenting with it. Can I feed it a pre-meshed yoke in order to speed up computation?

**Reply by OrtnerMichael** [reply]
no, but it would also not lead to a speedup. It uses the method of moments which relies on a dense matrix to solve, so you dont want to have large meshes anyways.
The principle idea is to create a Collection with a lot of magnets, and then `apply_demag` to compute all the interactions between all parts of the collection and change the magnetizations based on the given permeability.

The is a function included for the meshing.

**Reply by fdinhSG** [reply]
Got it. I'll try to download the module this week and play around with it. If I have any question I'll open a new thread. Thanks again!
